### ISSUE TITLE
Use case: do config symlinking during postinst

### ISSUE THREAD

**Issue by ehartsuyker** [sbt/sbt-native-packager#724]
Let's say for example I want to use Ansible to put some app specific configs on the host. I want them to go in `/etc/$appName/`. Dpkg won't copy the symlinked `/etc/` dir if it already exists. This leads to me having to manually symlink the `application.ini` to `/etc/$appName/` for each app.

Could the `postinst` script do `mkdir -p /etc/$appName` and then `ln` the config into it instead?

**Comment by muuki88** [comment]
Are you using the `JavaServerAppPackaging` archetype? This archetype links the `conf` directory to `/etc/$appName`. See [Server Archetype Docs](http://www.scala-sbt.org/sbt-native-packager/archetypes/java_server/index.html)

**Comment by ehartsuyker** [comment]
@muuki88 Yes. The symlink exists in the deb package (verified by looking in the `target` dir, and using `dpkg -x`), but if it's installed on a system when `/etc/$appName` already exists, the end result is that `application.ini` or whatever else may be in that dir don't end up under `/etc/$appName`. It's because (I think) dpkg see that the dir already exists and refuses to merge them.

**Comment by ehartsuyker** [comment]
Just to make it super obvious why this is a problem.

### Case 1
1. Install app
2. Render configs to `/etc/$appName`

`application.ini` is symlinked correctly because `dpkg` sees the dir doesn't exist and makes the new `/etc/$appName` dir. The symlink here points to the real file. But. The app breaks on start during `postinst` because of some missing `/etc/$appName/my-special-thing.conf`.

### Case 2
1. Render configs to `/etc/$appName`
2. Install app

I manually make the `/etc/$appName` dir and put `my-special-thing.conf` there so the app starts correctly. But. `dpkg` Doesn't want to overwrite the existent dirs, so the symlink in the package isn't created under `/etc/`, and thus `application.ini` doesn't appear under `/etc/$appName/`. I have to add a symlink manually.

Proposed solution: Instead of symlinking the dirs during the build, symlink each file in the `postinst` script before the app's start command.
